### PR TITLE
Fix rekey test

### PR DESCRIPTION
**Pull request by patrickxb**
- add notifyStart chan
- don't rely on fakeclock.BlockUntil

### CODE REVIEW

**Comment by patrickxb** [comment]
@maxtaco

**Comment by patrickxb** [comment]
cc @cjb

File: go/service/rekey_test.go [hunk 1]
```diff
@@ -109,13 +109,27 @@ func TestRekeyNeededMessageWithScores(t *testing.T) {
...
+	case <-time.After(20 * time.Second):
+		t.Fatal("timeout waiting for rekeyHandler.notifyStart")
+	}
+
+	done := make(chan struct{})
+	defer close(done)
 	go func() {
-		clock.BlockUntil(1)
-		clock.Advance(3 * time.Second)
+		for {
```
Comment (maxtaco) [inline go/service/rekey_test.go]: why is this loop needed?

Comment (patrickxb) [inline go/service/rekey_test.go]: to get the test to pass with a fake clock.  BlockUntil seems to be flakey.

**Comment by maxtaco** [comment]
So the bug was that the tester could run before the rekey process started up?

**Comment by patrickxb** [comment]
Yes.  The BlockUntil was supposed to wait until an After() call on a fake clock which would mean it got into the update loop, but despite many attempts I couldn't get it to work reliably.

With these changes, I haven't seen it fail.  Ran all rekey tests 10,000 times.

**Comment by maxtaco** [comment]
👍

**Comment by maxtaco** [comment]
also pkease unskip tests

**Comment by patrickxb** [comment]
They are unskipped.

**Pull request merged**